### DISCUSSION TITLE
New github user with manny errors

### DISCUSSION THREAD

**Discussion by PaulAniculaesei** [community/community#23401] [New to GitHub]
<p>Hello, community!I try to learn how to use git and Github and vs code.<br>
I clone a repository(not mine), I open it in vs code…I make a little change in “read me”, but I can’t commit and push it.<br>
The last error is : You don’t have permission to push to ‘name…’ on GitHub. Would you like to create a fork and push to it instead?<br>
Can you tell me why? Maybe someone can help me.<br>
I have tried for 6 hours but nothing…<br>
Thank you</p>

**Comment by blackjack4494** [comment]
<p>This is normal. If you clone a repository you simply download all the source and its history. However it seems the repository is not yours. You got a few options then. The most common one is to fork it (basically creating a copy of that repository or one of the branches). Since you forked it, the newly created repository is yours. If you want to contribute back to the ‘main’ project you can open up a Pull Request (PR) briefly explaining what you changed and why you think it would help the project.<br>
You simply just don’t get write access to any repository. Think about it. Literally anyone could spam or inject malicious code if it would be open to public by default.</p>

**Comment by PaulAniculaesei** [comment]
<p>Thank you for your time! But if I make from vs code a new branch , why I can’t open a pull request? It tells me I am not authorized to make changes…</p>

**Comment by PaulAniculaesei** [comment]
<p>For example…If I want to work on your repository…First I have to fork  it ,work on itt and then make a pull request?</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="PaulAniculaesei" data-post="3" data-topic="187771">
<div class="title">
<div class="quote-controls"></div>
 PaulAniculaesei:</div>
<blockquote>
<p>But if I make from vs code a new branch , why I can’t open a pull request?</p>
</blockquote>
</aside>
<p>Because you need to push the branch to somewhere public first. You can’t push your branch to someone else’s repository, so you need your own (a fork in GitHub terms).</p>

**Comment by blackjack4494** [comment]
<p>I suggest you check GitHub Docs <a href="https://docs.github.com/en/get-started/quickstart/fork-a-repo" class="inline-onebox" rel="noopener nofollow ugc">Fork a repo - GitHub Docs</a></p>
